src/web/components/ClientSideTokenGeneration/CstgAddDomainDialog.tsx

```diff
@@ -46,28 +59,9 @@ function CstgAddDomainDialog({
         message: 'The domains entered already exist.',
       });
     } else {
-      const invalidDomains: string[] = [];
-      let allValid = true;
-      uniqueDomains.forEach((newDomain) => {
-        if (!isValidDomain(newDomain)) {
-          invalidDomains.push(newDomain);
-          allValid = false;
-        }
-      });
-      if (!allValid) {
-        setError('root.serverError', {
-          type: '400',
-          message: `The domains entered are invalid: ${formatStringsWithSeparator(invalidDomains)}`,
-        });
-      } else {
-        // if all are valid but there are some non root-level domains, we make sure every domain is root-level
-        uniqueDomains.forEach((newDomain, index) => {
-          uniqueDomains[index] = extractTopLevelDomain(newDomain);
-        });
-        // filter for uniqueness (e.g. 2 different domains entered could have the same root-level domain)
-        const dedupedDomains = deduplicateStrings(uniqueDomains);
-        await onAddDomains(dedupedDomains, deleteExistingList);
-      }
+      // filter for uniqueness (e.g. 2 different domains entered could have the same root-level domain)
+      const dedupedDomains = deduplicateStrings(uniqueDomains);
+      await onAddDomains(dedupedDomains, deleteExistingList);
     }
```

Comment:
All the validation can be done on the back-end now

Reply:
I think this method is still pretty low, especially given that the validation can be done on the back-end now. Perhaps it might be worth refactoring this a bit to improve readability? Something like:
```
const onSubmit = async (formData: AddDomainNamesFormProps) => {
  const newDomains = separateStringsList(formData.newDomains);
  const uniqueDomains = getUniqueDomains(newDomains, deleteExistingList, existingDomains);

  if (uniqueDomains.length === 0) {
    handleError('The domains entered already exist.');
    return;
  }

  await handleValidDomains(uniqueDomains, deleteExistingList);
};

const getUniqueDomains = (
  newDomains: string[],
  deleteExistingList: boolean,
  existingDomains?: string[]
): string[] => {
  return deleteExistingList
    ? newDomains
    : newDomains.filter((domain) => !existingDomains?.includes(domain));
};

const handleError = (message: string) => {
  setError('root.serverError', {
    type: '400',
    message: message,
  });
};

const handleValidDomains = async (
  uniqueDomains: string[],
  deleteExistingList: boolean
) => {
  const dedupedDomains = deduplicateStrings(uniqueDomains);
  const invalidDomains = await onAddDomains(dedupedDomains, deleteExistingList);

  if (invalidDomains.length > 0) {
    handleError(
      `The domains entered are invalid root-level domains: ${formatStringsWithSeparator(
        invalidDomains
      )}`
    );
  }
};
```
(you don't need to use that verbatim)